packages/react-components/src/Grid.tsx

```diff
@@ -19,10 +29,30 @@ function Grid<TItem = GridDefaultItem>(
   props: GridProps<TItem>,
   ref: ForwardedRef<GridElement<TItem>>,
 ): ReactElement | null {
-  const [portals, rowDetailsRenderer] = useModelRenderer(props.rowDetailsRenderer, { renderSync: true });
+  const [portals, rowDetailsRenderer] = useModelRenderer(props.rowDetailsRenderer);
+
+  const innerRef = useRef<GridElement & { _recalculateColumnWidths?(...args: any[]): void }>(null);
+  const finalRef = useMergedRefs(innerRef, ref);
+
+  const [pendingRecalculateColumnWidthsCall, setPendingRecalculateColumnWidthsCall] = useState<any[] | null>(null);
+
+  useEffect(() => {
+    innerRef.current!._recalculateColumnWidths = function (...args) {
+      setPendingRecalculateColumnWidthsCall(args);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (pendingRecalculateColumnWidthsCall) {
+      const gridElement = innerRef.current!;
+      const gridProto = Object.getPrototypeOf(gridElement);
+      gridProto._recalculateColumnWidths.call(gridElement, ...pendingRecalculateColumnWidthsCall);
```

Comment:
Something isn't working with the `autoWidth` columns recalculation. I changed the Grid example on dev pages to:
```
...
<GridTreeColumn frozen path="name" header="This is a long header" autoWidth flexGrow={0} />
<GridColumn path="name" header="This is a long header" autoWidth flexGrow={0} />
...
```

On `main`:
![Screenshot 2024-11-18 at 11 52 41](https://github.com/user-attachments/assets/17aad656-2c84-425e-bbda-2954c09ac2ce)

On the PR branch:
![Screenshot 2024-11-18 at 11 52 57](https://github.com/user-attachments/assets/4d175f53-6513-4d36-a712-6078852bfc35)